modules/indexer/code/search.go

```diff
@@ -86,7 +87,22 @@ func HighlightSearchResultCode(filename, language string, lineNums []int, code s
 	return lines
 }
 
-func searchResult(result *internal.SearchResult, startIndex, endIndex int) (*Result, error) {
+func RawSearchResultCode(filename, language string, lineNums []int, code string) []*ResultLine {
+	// we should highlight the whole code block first, otherwise it doesn't work well with multiple line highlighting
+	rawLines := strings.Split(code, "\n")
+
+	// The lineNums outputted by highlight.Code might not match the original lineNums, because "highlight" removes the last `\n`
+	lines := make([]*ResultLine, min(len(rawLines), len(lineNums)))
```

Comment:
These comments appear to be copied from `HighlightSearchResultCode()` and are not relevant here (since we aren't using `highlight.Code()`).